spi_nand_flash/src/nand.c

```diff
@@ -289,6 +291,25 @@ esp_err_t spi_nand_erase_chip(spi_nand_flash_device_t *handle)
     return ret;
 }
 
+static bool s_need_data_refresh(spi_nand_flash_device_t *handle)
+{
+    uint8_t min_bits_corrected = 0;
+    bool ret = false;
+    if (handle->ecc_data.ecc_corrected_bits_status == STAT_ECC_1_TO_3_BITS_CORRECTED) {
+        min_bits_corrected = 1;
+    } else if (handle->ecc_data.ecc_corrected_bits_status == STAT_ECC_4_TO_6_BITS_CORRECTED) {
+        min_bits_corrected = 4;
+    } else if (handle->ecc_data.ecc_corrected_bits_status == STAT_ECC_7_8_BITS_CORRECTED) {
+        min_bits_corrected = 7;
+    }
+
+    // if number of corrected bits is greater than refresh threshold then rewite the sector
+    if (min_bits_corrected >= CONFIG_NAND_FLASH_ECC_DATA_REFRESH_THRESHOLD) {
```

Comment:
Seems like this is the only place where this option is used, so it's easy to make it a run-time configurable option (in spi_nand_flash_config_t). Kconfig options should be introduced only when changing something at run time is hard or expensive.